hamilton/graph.py

```diff
@@ -150,67 +151,233 @@ def create_graphviz_graph(
+        return f"<<table border=\"0\">{''.join(rows)}</table>>"
+
+    def _get_node_type(n: node.Node) -> str:
+        if (
+            n._node_source == node.NodeType.EXTERNAL
+            and n._originating_functions is None
+            and n._depended_on_by
+        ):
+            return "input"
+        elif (
+            n._node_source == node.NodeType.EXTERNAL
+            and n._originating_functions is None
+            and not n._depended_on_by
+        ):
+            return "config"
+        else:
+            return "function"
+
+    def _get_node_style(node_type: str):
+        fontname = "Helvetica"
+
+        if node_type == "config":
+            node_style = dict(
+                shape="note",
+                style="",
+                fontname=fontname,
+            )
+        elif node_type == "input":
+            node_style = dict(
+                shape="rectangle",
+                margin="0.15",
+                style="dashed",
+                fontname=fontname,
+            )
+        elif node_type == "materializer":
+            node_style = dict(
+                shape="cylinder",
+                margin="0.15,0.1",
+                fontname=fontname,
+            )
+        else:  # this is a function or else
+            node_style = dict(
+                shape="rectangle",
+                margin="0.15",
+                style="rounded,filled",
+                fillcolor="#b4d8e4",
+                fontname=fontname,
+            )
+
+        return node_style
+
+    def _get_function_modifier_style(modifier: str):
+        if modifier == "output":
+            modifier_style = dict(fillcolor="#FFC857")
+        elif modifier == "collect":
+            modifier_style = dict(peripheries="2", color="#EA5556")
+        elif modifier == "expand":
+            modifier_style = dict(peripheries="2", color="#56E39F")
+        elif modifier == "override":
+            modifier_style = dict(style="filled,diagonals")
+        elif modifier == "materializer":
+            modifier_style = dict(shape="cylinder")
+        else:
+            modifier_style = dict()
+
+        return modifier_style
+
+    def _get_edge_style(node_type: str):
+        if node_type == "expand":
+            edge_style = dict(
+                dir="both",
+                arrowhead="crow",
+                arrowtail="none",
+            )
+        elif node_type == "collect":
+            edge_style = dict(dir="both", arrowtail="crow")
+        else:
+            edge_style = dict()
+
+        return edge_style
+
+    def _get_legend(node_types):
+        legend_subgraph = graphviz.Digraph(
+            name="cluster_legend",  # needs to start with `cluster_` for graphviz layout
+            graph_attr=dict(
+                label="Legend",
+                rank="same",
+                fontname="helvetica",
+            ),
+        )
+
+        sorted_types = [
```

Comment:
I moved materializer downwards

It's a bespoke ordering, but I thought about having `config` and `input` first because they most often appear at the top of the graph near the legend. Then, all others are of `function` type with modifiers.

Reply:
👍